src/ripple/app/misc/Validations.cpp

```diff
@@ -104,39 +104,67 @@ class ValidationsImp : public Validations
         }
 
         auto hash = val->getLedgerHash ();
-        auto node = val->getNodeID ();
 
-        if (val->isTrusted () && isCurrent)
+        if (isCurrent &&
+            (val->isTrusted () || app_.validators ().listed (signer)))
         {
             ScopedLockType sl (mLock);
 
-            if (!findCreateSet (hash)->insert (std::make_pair (node, val)).second)
+            auto const masterKey = app_.manifestCache ().getMasterKey (signer);
```

Comment:
This should address the race condition:
https://github.com/ripple/rippled/pull/1842/commits/1132685ac847c116cdf551a0f9e31286a9069bdf